src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -525,6 +597,73 @@ public bool TryGetRestClients(string requestPath, [MaybeNullWhen(false)] out Has
             return _requestPathToResources;
         }
 
+        private string? GetDefaultNameFromConfiguration(OperationSet operationSet, ResourceTypeSegment resourceType)
+        {
+            if (_context.Configuration.MgmtConfiguration.RequestPathToResourceName.TryGetValue(operationSet.RequestPath, out var name))
+                return name;
+            if (_context.Configuration.MgmtConfiguration.RequestPathToResourceName.TryGetValue($"{operationSet.RequestPath}|{resourceType}", out name))
+                return name;
+
+            return null;
+        }
+
+        private string GetResourceName(string candidateName, OperationSet operationSet, RequestPath requestPath)
+        {
+            // read configuration to see if we could get a configuration for this resource
+            var resourceType = requestPath.GetResourceType(_mgmtConfiguration);
```

Comment:
I have a feeling that the `requestPath` here is only constructed from the path segments, therefore it must be a "unexpanded" path, therefore the resource type from this path directly will have variables in it.
We might need to find a way to pin this to a specific resource type. But how? I did not think of a way from the information we have here.

Reply:
this is the expanded resource type